packages/field-plugin/src/createFieldPlugin/FieldPluginActions.ts

```diff
@@ -1,7 +1,8 @@
 import { Asset } from '../messaging'
 
-export type SetContent = (content: unknown) => void
-export type SetModalOpen = (isModal: boolean) => void
+export type DispatchAction<T> = T | ((value: T) => T)
+export type SetContent = <C>(setContentAction: DispatchAction<C>) => void
```

Comment:
```suggestion
export type DispatchAction<T> = (value: T) => T
export type SetContent = <C>(content: C | DispatchAction<C>) => void
```

What about this?

Reply:
I copied the `DispatchAction` from @types/react

Reply:
Ok, hang on, React actually calls it `SetStateActions`. I will rename it to this